added backbone for future RAG implementation

ogbozoyan opened the pull request.

Mikhail-Lytov commented on src/main/kotlin/ru/ogbozoyan/cron/web/controller/CronController.kt:
```diff
@@ -1,21 +1,45 @@
+    @Operation(
+        summary = "Validate CRON expression",
+        description = "Validates the provided CRON expression and returns the result"
+    )
```
add examples of responses

Mikhail-Lytov reviewed (changes requested):
Add documentation

ogbozoyan commented:
> Add documentation

will be added sooner

ogbozoyan committed:
add: model size

ogbozoyan committed:
add: model size

ogbozoyan committed:
add: model size

ogbozoyan committed:
add: postgres, system-message.st, model
draft:  RAG by adding embeding model, added test function of callable fuctions for ai

ogbozoyan committed:
add: postgres, system-message.st, model
draft:  RAG by adding embeding model, added test function of callable fuctions for ai

ogbozoyan committed:
add: postgres, system-message.st, model
draft:  RAG by adding embeding model, added test function of callable fuctions for ai

ogbozoyan committed:
add: howto debugging setup

ogbozoyan committed:
renamed chatClint bean

ogbozoyan committed:
removed lombok and refactored pgvector provider in docker compose

ogbozoyan committed:
add: migration for init vector store
feat: add check relevance

ogbozoyan committed:
add: migration for init vector store
feat: add check relevance

ogbozoyan committed:
add: migration for init vector store
feat: add check relevance

Commit:
fix

Signed-off-by: ogbozoyan <oganes.bozoyan@mail.ru>
